Contributing - update licensing and tips (#871)
ARMmbed/mbed-os-5-docs: docs/reference/contributing/guidelines/workflow.md

#### Fix

> A bug fix is a change that fixes a specific defect in the codebase with backwards compatibility. These are the highest priority because of the positive impact the change will have on users developing against the same code. A bug fix should be limited to restoring the documented or proven otherwise, originally intended behavior. Bug fixes are candidates for patch releases.

Reviewer: ```suggestion
A bug fix is a change that fixes a specific defect in the codebase with backwards compatibility. These are the highest priority because of the positive impact the change will have on users developing against the same code. A bug fix should be limited to restoring the documented or proven otherwise, originally intended behavior. Bug fixes are candidates for patch releases. Large, non-trivial bug fixes approaching the size of refactors run the risk of being considered refactors themselves.

@0xc0170 

```

Author: +1, added

Release: patch

#### Refactor

> A refactor is a contribution that modifies the codebase without fixing a bug or changing the existing behavior. Examples of this would be moving functions or variables between translation units, renaming source files or folders, scope modification for nonpublic code, documentation structure changes, and test organization changes. There is always the risk that someone depended on the location or name before a refactor therefore these are lower in priority than bug fixes and might require detailed justification for the change. Refactors are candidates for feature releases. 

Reviewer: If a refactor is made in a backwards-compatble way, can they be targeted for patch releases?

Reviewer: What about if a refactor fixes a bug?

Author: Either they fix a bug separately or will go with refactor. Everything should be backward compatible (only exceptionally not)

Reviewer: So making sure, if a the only way to fix a particular bug is with a refactor, that PR will have to come into the next major release?

Author: It should

Release: patch

#### Functionality change

> Any change in the functionality, it can be adding new feature or adding new method or function. It does not matter if it is C++, C or Python.

Reviewer 2: > adding new feature or adding new method or function

adding a new feature, adding a new method or a function

Author: Thanks , will fix all !

> Any change in the functionality, it can be adding a new feature, adding a new method or a function. It does not matter if it is C++, C or Python.

Reviewer: ```suggestion
Any change in the functionality, it can be adding a new feature, adding a new method or a function. Software language does not matter.
```

Reviewer: What about changes required to bring in new targets?
